test/jaxp/javax/xml/jaxp/unittest/transform/PropertiesTest.java

```diff
@@ -0,0 +1,100 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/*
+ * @test
+ * @bug 8344925
+ * @summary Transformer properties tests
+ * @library /javax/xml/jaxp/libs /javax/xml/jaxp/unittest
+ * @run junit/othervm transform.PropertiesTest
+ */
+public class PropertiesTest {
+    private static final String USER_DIR = System.getProperty("user.dir");
+    private static final String TEST_DIR = System.getProperty("test.src");
+    // Test parameters:
+    // generate-translet: indicates whether to generate translet
+    // translet-name: the name of the translet
+    // package-name: the package name
+    // destination-directory: the destination
+    // expected: the class path
+    private static Stream<Arguments> testData() {
+        String destination = JUnitTestUtil.CLS_DIR + "/testdir";
+        return Stream.of(
+                Arguments.of(true, "MyTranslet", "org.myorg", destination, "/org/myorg/MyTranslet.class"),
+                Arguments.of(false, "Translet", "not.generate", destination, "/not/generate/Translet.class"),
+                // translet named after the stylesheet
+                Arguments.of(true, null, "org.myorg", destination, "/org/myorg/transform.class"),
+                // default package name die.verwandlung since JDK 9
+                Arguments.of(true, "MyTranslet", null, destination, "/die/verwandlung/MyTranslet.class"),
+                Arguments.of(true, "MyTranslet", "org.myorg", null, "/org/myorg/MyTranslet.class")
+                );
+    }
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        // so that the translet is generated under test.classes
+        JUnitTestUtil.copyFile(JUnitTestUtil.SRC_DIR + "/transform.xsl", JUnitTestUtil.CLS_DIR + "/transform.xsl");
+    }
+
+    @ParameterizedTest
+    @MethodSource("testData")
+    public void test(boolean generateTranslet, String name, String packageName,
+                     String destination, String expected)
+            throws Exception {
+        TransformerFactory tf = TransformerFactory.newInstance();
+
+        tf.setAttribute("generate-translet", generateTranslet);
+        if (name != null) tf.setAttribute("translet-name", name);
+        if (packageName != null) tf.setAttribute("package-name", packageName);
+        if (destination != null) tf.setAttribute("destination-directory", destination);
+
+        String xslFile = JUnitTestUtil.CLS_DIR + "/transform.xsl";
+        String xslSysId = JUnitTestUtil.getSystemId(xslFile);
+        StreamSource xsl = new StreamSource(xslSysId);
+        tf.newTemplates(xsl);
+
+        String path = (destination != null) ? destination + expected : new File(xslFile).getParent() + expected;
+
+        if (generateTranslet) {
+            //Files.list(Path.of(path)).forEach(System.out::println);
```

Comment:
Left over from testing the test? I'd suggest to remove this commented out code.

Reply:
Removed. Thanks.